src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    try {
+      if (!isComponentInstalled("app-engine-java")) {
```

Comment:
Could we use `JAVA_TOOLS_JAR` as a heuristic: if the path exists, then we seem to have `app-engine-java` installed, otherwise confirm with `gcloud components list`.